webui/react/src/pages/Authentication.tsx

```diff
@@ -0,0 +1,156 @@
+  if (isLogout) {
+    logout({});
+    setAuth({ type: Auth.ActionType.Reset });
+    history.push('/det/login' + props.location.search);
+  }
+
+  const onFinish = (creds: FromValues): void => {
+    // TODO validate the creds type?
+    setCanSubmit(false);
+    const hideLoader = message.loading('logging in..');
+    login(creds as Credentials)
+      .then(() => updateAuth(setAuth))
+      .catch((e: Error) => {
+        setBadCredentials(isLoginFailure(e));
+        if (badCredentials) return;
+        // DISCUSS do we want to potentially report it?
+        // or pass it through error handler for some reason?
+        const actionMsg = badCredentials ? 'check your username and password.' : 'retry.';
+        handleError({
+          error: e,
+          isUserTriggered: true,
+          message: e.message,
+          publicMessage: `Failed to login. Please ${actionMsg}`,
+          publicSubject: 'Login failed',
+          type: badCredentials ? ErrorType.Input : ErrorType.Server,
+        });
+      })
+      .finally(() => {
+        setCanSubmit(true);
+        hideLoader();
+      });
+  };
+
+  const onValuesChange = (changes: FromValues, values: FromValues): void => {
+    const hasUsername = !!values.username;
+    setBadCredentials(false);
+    setCanSubmit(hasUsername);
+  };
+
+  const loginForm = (
+    <Form
+      name="basic"
+      size="large"
+      onFinish={onFinish}
+      onValuesChange={onValuesChange}
+    >
+      <Form.Item
+        name="username"
+        rules={[
+          {
+            message: 'Please input your username!',
+            required: true,
+          },
+        ]}
+      >
+        <Input placeholder="Username" prefix={<Icon name="user" />} />
+      </Form.Item>
+
+      <Form.Item name="password">
+        <Input.Password placeholder="Password" prefix={<Icon name="lock" />} />
+      </Form.Item>
+
+      {
+        badCredentials &&
+          <p className={css['error-message']}>
+            Incorrect username or password, please try again.
+          </p>
+      }
+
+      <Form.Item>
+        <Button disabled={!canSubmit} htmlType="submit" type="primary">
+          Sign In
+        </Button>
+      </Form.Item>
+    </Form>
+
+  );
+
+  if (isLogout || isLoading) return <Spinner fullPage />;
+  if (auth.isAuthenticated) {
+    if (isCrossoverRoute(redirectUri)) {
+      crossoverRoute(redirectUri);
+      return <Spinner fullPage />;
+    }
+    return <Redirect to={redirectUri} />;
+  }
+
+  return (
+    <div className={css.base}>
+      <div className={css.content}>
+        {/* DISCUSSION what if we didn't need to add the logo classname and was able to
+        target logo on its own using component name easily in module.scss */}
+        <Logo className={css.logo} type={LogoTypes.Dark} />
+        {loginForm}
+        <a href="/docs/system-administration/users.html?highlight=user" target="_blank">
+          Forgot your password or need to create a user? Checkout our docs
```

Comment:
`Check out`
Also, could we make only the `docs` a hyperlink? Or does that not look great?

Reply:
that works